pages/organizations/[organizationSlug]/activity.tsx

```diff
@@ -0,0 +1,328 @@
+            Created record{" "}
+            <Code>
+              <a href={urlToRecord}>{activity.recordId}</a>
+            </Code>{" "}
+            in{" "}
+            <Code>
+              <a href={urlToSource}>{sourceName}</a>
+            </Code>
+          </p>
+        );
+      case "bulkDelete":
+        return (
+          <p>
+            Deleted{" "}
+            <Tooltip label={activity.recordId} placement="auto">
+              <u>multiple records</u>
+            </Tooltip>{" "}
+            from{" "}
+            <Code>
+              <a href={urlToSource}>{sourceName}</a>
+            </Code>
+          </p>
+        );
+      case "delete":
+        return (
+          <p>
+            Deleted record <Code>{activity.recordId}</Code> from{" "}
+            <Code>
+              <a href={urlToSource}>{sourceName}</a>
+            </Code>
+          </p>
+        );
+      case "update":
+        return (
+          <p>
+            Updated record{" "}
+            <Code>
+              <a href={urlToRecord}>{activity.recordId}</a>
+            </Code>{" "}
+            in{" "}
+            <Code>
+              <a href={urlToSource}>{sourceName}</a>
+            </Code>
+          </p>
+        );
+    }
+  }, [activity.action]);
+
+  const getChangeValue = (change: any) => {
+    if (isUndefined(change))
+      return <span className="text-xs uppercase">Undefined</span>;
+    if (isNull(change)) return <span className="text-xs uppercase">Null</span>;
+    if (isEmpty(change))
+      return <span className="text-xs uppercase">Empty</span>;
+
+    if (typeof change === "object") return JSON.stringify(change, null, 2);
+
+    return change;
+  };
+
+  const activityChanges: Record<string, any>[] = useMemo(() => {
+    return activity.changes as Record<string, any>[];
+  }, [activity.changes]);
+
+  const changes = useMemo(() => {
+    if (activityChanges.length > 0) {
+      return (
+        <>
+          <span className="text-xs font-bold uppercase text-gray-800 leading-none">
+            Changes
+          </span>
+          <ul className="space-y-1">
+            {activityChanges &&
+              activityChanges.map(
+                (change: Record<string, any>, idx: number) => {
+                  return (
+                    <li key={idx}>
+                      <span className="text-xs mr-1 font-semibold">
+                        {change.column}
+                      </span>{" "}
+                      <br />
+                      <span className="text-gray-700">
+                        {getChangeValue(change.before)}
+                      </span>{" "}
+                      <ArrowRightIcon className="h-3 w-3 inline mb-1 mx-1" />{" "}
+                      <span className="text-gray-900">
+                        {getChangeValue(change.after)}
+                      </span>
+                    </li>
+                  );
+                }
+              )}
+          </ul>
+        </>
+      );
+    }
```

Comment:
Could you extract this to a component too? The component presents the content. Then you don't even need to memorize it.